unraid_mcp/tools/unraid.py

```
@@ -0,0 +1,1910 @@
+    all_live = set(SNAPSHOT_ACTIONS) | set(COLLECT_ACTIONS)
+    if subaction not in all_live:
+        raise ToolError(
+            f"Invalid subaction '{subaction}' for live. Must be one of: {sorted(all_live)}"
+        )
+
+    if subaction == "log_tail":
+        if not path:
+            raise ToolError("path is required for live/log_tail")
+        normalized = os.path.realpath(path)  # noqa: ASYNC240
+        if not any(normalized.startswith(p) for p in _LIVE_ALLOWED_LOG_PREFIXES):
+            raise ToolError(f"path must start with one of: {', '.join(_LIVE_ALLOWED_LOG_PREFIXES)}")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)